fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/AgilityContext.java

```diff
@@ -249,6 +239,20 @@ public void commitNow() {
             }
         }
 
+        private <T> T reportAndReThrowException(Throwable ex) {
+            if (LOGGER.isDebugEnabled()) {
+                long nowMilliseconds = now();
+                final long creationAge = nowMilliseconds - creationTime;
+                final long  prevCheckAge = nowMilliseconds - prevCommitCheckTime;
+                LOGGER.debug(KeyValueLogMessage.build("AgilityContext: Commit failed",
+                        LogMessageKeys.AGILITY_CONTEXT_AGE_MILLISECONDS, creationAge,
+                        LogMessageKeys.AGILITY_CONTEXT_PREV_CHECK_MILLISECONDS, prevCheckAge,
+                        LogMessageKeys.AGILITY_CONTEXT_WRITE_SIZE_BYTES, currentWriteSize
+                ).toString(), ex);
+            }
+            throw new AssertionError(ex); // re-throw
```

Comment:
You shouldn't convert this into an `AssertionError`, it should still be whatever error it was. Throwing an `AssertionError` could mess with exception handling, as it is not expected that code would throw an `AssertionError`, unless `assert`s are enabled, which they rarely are in production environments.
From the `Error` [docs](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Error.html):
> An Error is a subclass of Throwable that indicates serious problems that a reasonable application should not try to catch.

Reply:
✅